ObjectState[] object_state_list